code/modules/mining/machine_vending.dm

```diff
@@ -263,6 +263,12 @@
 	if(prob(50 / severity) && severity < 3)
 		qdel(src)
 
+/obj/machinery/mineral/equipment_vendor/Destroy()
+	if(inserted_id)
+		inserted_id.forceMove(loc)
+	..()
```

Comment:
`return ..()`
Destroy returns a value which is used to determine what happens next to the object. If it gets queued in the garbage collector. Or if it gets hard deleted etc

Reply:
I thought that'd get called anyway, since `..()` is there.

Change it to `return ..()`, then?

Reply:
only calling `..()` will just call the parent. if you do `return ..()` it will call the parent and return the value the parent returns. So aye aye